python/src/dataset.rs

```diff
@@ -180,6 +183,40 @@ impl Dataset {
             Ok(pyvers)
         })
     }
+
+    fn create_index(
+        self_: PyRef<'_, Self>,
+        columns: Vec<&str>,
+        index_type: &str,
+        name: Option<String>,
+        kwargs: &PyDict,
+    ) -> PyResult<()> {
+        let idx_type = match index_type.to_uppercase().as_str() {
```

Comment:
do we need to add a check that the vector column is fixed size array of float32?

Reply:
The check was done in the core `/rust` side.